csm/common/timeseries.py

```diff
@@ -270,15 +270,20 @@ async def _parse_interval(self, from_t, duration_t, interval, total_sample):
         duration_t = int(duration_t) - int(self._offset_interval)
         if total_sample == "" and interval == "":
             interval = str(self._storage_interval) + 's'
-        elif total_sample != "":
+        elif total_sample != "" and interval == "":
             try:
                 interval = str(int(diff_sec / int(total_sample))) + 's'
             except (ValueError, ZeroDivisionError):
                 raise InvalidRequest("'total_sample' should be integer and greater than zero")
         elif interval != "":
-            interval = str(int(interval)) + 's'
-        else:
-            raise CsmInternalError("Unable to parse interval")
+            try:
+                interval = int(interval)
+            except ValueError:
+                raise InvalidRequest("'interval' should be integer and greater than zero or blank")
+            if interval < 1:
+                raise InvalidRequest("'interval' should be integer and greater than zero or blank")
+            else:
+                interval = str(int(interval)) + 's'
```

Comment:
If this statement fails return InvalidRequest error.

Reply:
Addressed

Reply:
You can do like this - 
try:
	interval = int(interval)
	if interval < 1:
		raise InvalidRequest("'interval' should be integer and greater than zero.")
	interval = str(interval) + 's'
except (ValueError, TypeError):
	raise InvalidRequest("'interval' should be integer and greater than zero.")
   
Also, one thought if you are checking for lower value like -ve then you should check for higher value too. Like what if I give 99898988889998889998 as a value?
For now its OK.

Reply:
> You can do like this -
> try:
> interval = int(interval)
> if interval < 1:
> raise InvalidRequest("'interval' should be integer and greater than zero.")
> interval = str(interval) + 's'
> except (ValueError, TypeError):
> raise InvalidRequest("'interval' should be integer and greater than zero.")
> 
> Also, one thought if you are checking for lower value like -ve then you should check for higher value too. Like what if I give 99898988889998889998 as a value?
> For now its OK.

Addressed, I had asked to Pranay for Higher value, but he said that there is no higher value limit, that why did not check that.